lib/dialects/sqlite/query-generator.js

```diff
@@ -79,6 +81,153 @@ const QueryGenerator = {
     return !!value ? 1 : 0;
   },
 
+  // A recursive parser for nested where conditions
+  // (copied from ../postgres/query-generator.js)
+  parseConditionObject(_conditions, path) {
```

Comment:
If this same method being used in both dialects, we should move it to AbstractQueryGen